server/plugin/plugin.go

```diff
@@ -376,14 +379,59 @@ func (p *Plugin) CreateBotDMPost(userID, message, postType string) {
 	}
 }
 
+func (p *Plugin) CheckISDuplicateSummary(userName, text string) (bool, error) {
+	p.API.LogWarn("checking CheckISDuplicateSummary", "userName", userName)
+	previousSummary, err := p.GetDailySummaryText(userName)
+	p.API.LogWarn("checking previousSummary", "previousSummary", previousSummary)
+	if err != nil {
+		return false, err
+	}
+	if previousSummary == text {
+		return true, nil
+	}
+	return false, nil
+}
+func (p *Plugin) StoreDailySummaryText(userName, summaryText string) (bool, error) {
+	if err := p.API.KVSet(userName+dailySummary, []byte(summaryText)); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+func (p *Plugin) GetDailySummaryText(userName string) (string, error) {
+	summaryByte, err := p.API.KVGet(userName + dailySummary)
+	if err != nil {
+		return "", err
+	}
+	if summaryByte == nil {
+		return "", nil
+	}
+	p.API.LogWarn("GetDailySummaryText", "summary text ", string(summaryByte))
+	return string(summaryByte), nil
+}
+
 func (p *Plugin) PostToDo(info *GitHubUserInfo) {
 	text, err := p.GetToDo(context.Background(), info.GitHubUsername, p.githubConnect(*info.Token))
 	if err != nil {
 		p.API.LogWarn("Failed to get todo text", "userID", info.UserID, "error", err.Error())
 		return
 	}
-
-	p.CreateBotDMPost(info.UserID, text, "custom_git_todo")
+	if info.Settings.DailyReminder {
```

Comment:
Should this check `OnChange`?

Reply:
resolved , Sorry my bad it got changed before pushing 
it should be onChange